autoscheduler/frontend/src/components/SchedulingPage/CourseSelectColumn/CourseSelectCard/ExpandedCourseCard/SectionSelect/SectionSelect.tsx

```diff
@@ -73,43 +157,87 @@ const SectionSelect: React.FC<SectionSelectProps> = ({ id }): JSX.Element => {
     );
   }
 
-  let numSelected = 0;
   /**
    * Makes a list of `SectionInfo` elements, one for each section of this course, by iterating over
    * each section in `sections`. As it iterates, this function groups consecutive sections with the
    * same professor and honors status together inside one `<ul>` and under one header. Having them
    * all inside the same `<ul>` is important in order to get smooth transitions with sticky headers.
    */
+  let allSelected = true;
   const makeList = (): JSX.Element[] => {
+    /**
+     * Function to filter based off of selected filters.
+     * We need to use this here and in professor group,
+     * since professor group only gets a start and end index.
+     */
+    const filterSections = (sectionData: SectionSelected): boolean => {
+      const toBool = (filter: SectionFilter | undefined, val: boolean): boolean => {
```

Comment:
```suggestion
    const shouldIncludeSection = (sectionData: SectionSelected): boolean => {
      // Whether the section should be included based on a SectionFilter and its value
      const filterOnVal = (filter: SectionFilter | undefined, val: boolean): boolean => {
```
I don't like the name `filterSections`, since it only checks a single section (whereas the name makes it seem like it should filter all of them at once), try using the new name I provided (the same applies to the `ProfessorGroup component's property with the same name).

Similarly `toBool` isn't a very good name: it's hard to tell what it does: changing the name and adding a comment will help with readability

Reply:
I probably shouldn't have made this a suggestion because you need to change the usage of `toBool`, it broke the most recent build. I'm unresolving this just to make a note to fix it

Reply:
Oh sorry I fixed that already, didn't push yet